### DISCUSSION TITLE
User resetting their password does not trigger "OnUserBeforeUpdateRequest" hook

### DISCUSSION THREAD

**Discussion by KyungWonPark** [pocketbase/pocketbase#946] [Q&A] [answered]
Hello

When I (admin) update a user's password from admin dashboard it triggers "OnUserBeforeUpdateRequest" hook, but users resetting their password through email link does not trigger "OnUserBeforeUpdateRequest" hook.

How can I add custom hook logic to the event that users resetting their password?

Thank you

**Comment by ganigeorgiev** [comment, marked as answer]
The `OnUserBeforeUpdateRequest` is triggered only `PATCH /api/users/:id` request and in v0.7.x only admins are allowed to perform this action. In v0.8.0-rc this is no longer the case and user can change their password directly too if they provide `oldPassword` (see [v0.8.0-rc1 release notes](https://github.com/pocketbase/pocketbase/releases/tag/v0.8.0-rc1)).

Email password reset confirmation requests don't trigger this hook. Instead you can attach to the more generic hook that listens to all db model changes, aka. `OnModelBeforeUpdate`:

```go
// v0.7.x
app.OnModelBeforeUpdate().Add(func(e *core.ModelEvent) error {
	if (e.Model.TableName() != "_users") {
		return nil // not an user update
	}

	// do something with the updated user model
	user := e.Model.(*models.User)

	oldUser, err := app.Dao().FindUserById(user.Id)
	if err != nil {
		return err
	}

	if oldUser.PasswordHash != user.PasswordHash {
	    // password changed..
	}

	return nil
})

// v0.8.x
app.OnModelBeforeUpdate().Add(func(e *core.ModelEvent) error {
	record, ok := e.Model.(*models.Record)
	if !ok || record.Collection().Name != "users" {
		return nil // not an user update
	}

	oldRecord, err := app.Dao().FindRecordById("users", record.Id)
	if err != nil {
		return err
	}

	// currently there is no getter PasswordHash(), but I'll add it before the final v0.8.0
	if oldRecord.GetString("passwordHash") != record.GetString("passwordHash") {
	    // password changed...
	}

	return nil
})
```

_Edit: I've updated the example to show how you could detect password change._
